sw/device/lib/testing/csrng_testutils.c

```diff
@@ -14,64 +14,64 @@ enum {
   kNumOutputWordsMax = 16,
 };
 
-void csrng_testutils_cmd_ready_wait(const dif_csrng_t *csrng) {
+status_t csrng_testutils_cmd_ready_wait(const dif_csrng_t *csrng) {
   dif_csrng_cmd_status_t cmd_status;
   do {
-    CHECK_DIF_OK(dif_csrng_get_cmd_interface_status(csrng, &cmd_status));
-    CHECK(cmd_status.kind != kDifCsrngCmdStatusError);
+    TRY(dif_csrng_get_cmd_interface_status(csrng, &cmd_status));
+    TRY_CHECK(cmd_status.kind != kDifCsrngCmdStatusError);
   } while (cmd_status.kind != kDifCsrngCmdStatusReady);
+  return OK_STATUS();
 }
 
-void csrng_testutils_cmd_generate_run(const dif_csrng_t *csrng,
-                                      uint32_t *output, size_t output_len) {
-  csrng_testutils_cmd_ready_wait(csrng);
-  CHECK_DIF_OK(dif_csrng_generate_start(csrng, output_len));
+status_t csrng_testutils_cmd_generate_run(const dif_csrng_t *csrng,
+                                          uint32_t *output, size_t output_len) {
+  TRY(csrng_testutils_cmd_ready_wait(csrng));
+  TRY(dif_csrng_generate_start(csrng, output_len));
 
   dif_csrng_output_status_t output_status;
   do {
-    CHECK_DIF_OK(dif_csrng_get_output_status(csrng, &output_status));
+    TRY(dif_csrng_get_output_status(csrng, &output_status));
   } while (!output_status.valid_data);
 
-  CHECK_DIF_OK(dif_csrng_generate_read(csrng, output, output_len));
+  TRY(dif_csrng_generate_read(csrng, output, output_len));
+  return OK_STATUS();
 }
 
-void csrng_testutils_check_internal_state(
+status_t csrng_testutils_check_internal_state(
     const dif_csrng_t *csrng, const dif_csrng_internal_state_t *expected) {
-  csrng_testutils_cmd_ready_wait(csrng);
+  TRY(csrng_testutils_cmd_ready_wait(csrng));
   dif_csrng_internal_state_t got;
-  CHECK_DIF_OK(
-      dif_csrng_get_internal_state(csrng, kCsrngInternalStateIdSw, &got));
+  TRY(dif_csrng_get_internal_state(csrng, kCsrngInternalStateIdSw, &got));
 
-  CHECK(got.instantiated == expected->instantiated);
-  CHECK(got.reseed_counter == expected->reseed_counter);
-  CHECK(got.fips_compliance == expected->fips_compliance);
+  TRY_CHECK(got.instantiated == expected->instantiated);
+  TRY_CHECK(got.reseed_counter == expected->reseed_counter);
+  TRY_CHECK(got.fips_compliance == expected->fips_compliance);
 
-  CHECK_ARRAYS_EQ(got.v, expected->v, ARRAYSIZE(expected->v),
-                  "CSRNG internal V buffer mismatch.");
+  TRY_CHECK(memcmp(got.v, expected->v, sizeof(expected->v)) == 0);
```

Comment:
Should `TRY_CHECK(memcmp(...))` be turned into something like `TRY_CHECK_ARRAYS_EQ(...)`?

Reply:
If we want the logs yes, otherwise `memcmp` would be fine.

Reply:
Since we may want to move the LOG*s out of the INTERNAL_CHECK macro eventually if we don't want a bunch of unformatted log messages printed out of the console, interleaved with well-formatted (i.e., JSON serialized) messages, as I mentioned [below](https://github.com/lowRISC/opentitan/pull/17563#discussion_r1140479888), this may not matter.